lib/icingadb/redisconnection.cpp

```diff
@@ -340,7 +334,7 @@ void RedisConnection::Connect(asio::yield_context& yc)
 					}
 
 					Handshake(conn, yc);
-					waitForReadLoop();
+					m_NoQueuedReads.Wait(yc);
```

Comment:
I often struggle to differentiate between the `AsioConditionVariable::Clear()` and `AsioConditionVariable::Set()` methods since they seem to perform very similar functions, and neither of them includes a clear comment explaining their purpose. And yet, I had to trace on my own to understand why the `init` parameter is set for this in the constructor but not for the others. Could you please provide some comments explaining the scenarios in which this gets unblocked? Additionally, I would suggest naming that new property something like `m_ReadLoopDone` to better reflect its original intentions.